authz/providers/azure/rbac/rbac.go

```diff
@@ -375,7 +378,94 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 
 		finalStatus = status
 	}
-	return finalStatus, nil
+	if finalStatus != nil && finalStatus.Allowed {
+		klog.V(5).Infof("Checkaccess request is allowed for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	if !a.useManagedNamespaceResourceScopeFormat || a.clusterType != managedClusters {
+		klog.V(5).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	klog.V(5).Infof("Falling back to checking managed namespace scope", checkAccessUsername)
```

Comment:
Should we extract all this new logic to a new sub function ?

Reply:
Because this code is on a really critical path, we didn't want to do anything that can potentially cause any regression (or a security incident) for the existing behavior. That's why I didn't extract/refactor.